Assets/Scripts/UI/Components/TextTicker.cs

```diff
@@ -0,0 +1,107 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+namespace Netherlands3D.Twin
+{
+    public class TextTicker : MonoBehaviour
+    {
+        public RectTransform childRect; // Assign the child RectTransform in the inspector  
+        public float scrollSpeed = 200f; // Speed of scrolling
+
+        private RectTransform parentRect;
+        private float childWidth;
+        private float parentWidth;
+        private Coroutine scrollingCoroutine;
+
+        void Awake()
+        {
+            parentRect = GetComponent<RectTransform>();
+        }
+
+        void Start()
+        {
+            // Get initial widths and set initial position  
+            UpdateWidths();
+            SetInitialPosition();
+
+            // You can call this method to start monitoring for width changes  
+            StartMonitoringWidth();
+        }
+
+        private void SetInitialPosition()
+        {
+            if (childWidth > parentWidth)
+            {
+                childRect.anchoredPosition = new Vector2(0, childRect.anchoredPosition.y);
+            }
+        }
+
+        private void UpdateWidths()
+        {
+            // Get widths of parent and child RectTransforms  
+            childWidth = childRect.rect.width;
+            parentWidth = parentRect.rect.width;
+        }
+
+        // Start monitoring for width changes  
+        public void StartMonitoringWidth()
+        {
+            UpdateWidths(); // Initial check 
+            CheckAndStartScrolling(); // Check and start scrolling if necessary  
+        }
+
+        // Example event trigger method to be called when needed  
+        public void OnWidthChange() // Call this method when a width change occurs  
+        {
+            Invoke("UpdateWidths",0.1f);
+            Invoke("CheckAndStartScrolling",0.1f);
```

Comment:
je zou evt een timer method kunnen schrijven met coroutine om te voorkomen dat als method namen veranderen in de toekomst er geen literals zijn die vergeten worden, maar voor nu prima